code/graph_algorithms/src/lowest_common_ancestor/lowest_common_ancestor.cpp

```diff
@@ -0,0 +1,82 @@
+#include <iostream>
+#include <vector>
+#include <array>
+using namespace std;
+
+// Part of Cosmos by OpenGenus Foundation
+
+// Problem Statement
+/*
+ * Given a tree with n nodes
+ * Your task is to process q queries of the following form:
+ * who is the lowest common ancestor of nodes a and b in the tree?
+ *
+ * The lowest common ancestor of two nodes of a rooted tree is the lowest node
+ * whose subtree contains both the nodes.
+ */
+
+void dfs(int k, int p, vector<int> &depth, vector<vector<int>> &adj, vector<array<int, 20>> &ancestor)
+{
+    ancestor[k][0] = p;
```

Comment:
Maybe I'm wrong, but I think we are not considering here the case in which k is the root node.

I would do:

`if (p == 0) {
        depth[k] = 0; // Root node
        ancestor[k][0] = k; // Root node's ancestor is itself
    } else {
        depth[k] = depth[p] + 1;
        ancestor[k][0] = p;
    }`
    

What do you think?

Reply:
I don't think it is necessary because the default values for arrays are already zero.

Reply:
Yes completely agree, I missed the implicit initialization, nice job by the way!